counter/src/index.ts

```diff
@@ -1,3 +1,20 @@
-export * from "./counterAPI";
-export * from "./Counter";
-export * from "./counterSlice";
+import { useDispatch, useSelector } from "react-redux";
+import { Counter } from "./Counter";
+import { counterSliceReducer, initializeSlicePackage, counterSliceName, selectCount, incrementByAmount } from "./counterSlice";
+
+/*
+    Ideally, these functions would live in some `counterSliceApi` module.
+*/
+
+function useCurrentCountValue() {
+    const currentCountValue = useSelector(selectCount)
+    return currentCountValue
+}
+function useCountIncrementor() {
```

Comment:
Does this need to be a hook?  could it just wrap `dispatch` instead of `useDispatch` and be a normal function or would that create some problem i'm forgetting?

Reply:
Not sure what you mean. Are you thinking we take the result of `useDispatch` as an argument instead of calling `useDispatch`? In hindsight, I probably should have used `useSliceDispatch` and `useSliceSelector`.

e.g.
```ts
function createCountIncrementor(dispatch: ReturnType<typeof useDispatch>) {
    return (amount: number) => {
        dispatch(incrementByAmount(amount))
    }
}
```

Reply:
Right; that's what I was wondering.  Would that work?